scripts/faq_script.ts

```diff
@@ -0,0 +1,183 @@
+  }[];
+}
+
+async function uploadImage(imageUrl: string) {
+  const filesService = new FilesService();
+
+  const fetchedImage = await fetch(imageUrl);
+  if (!fetchedImage.ok) {
+    throw new Error(
+      `Failed to download image. HTTP status: ${fetchedImage.status}`,
+    );
+  }
+
+  const extension = fetchedImage.headers.get("content-type")?.split("/")[1];
+  const arrayBuffer = await fetchedImage.arrayBuffer();
+  const buffer = Buffer.from(arrayBuffer);
+  const size = buffer.length;
+
+  const file = {
+    size,
+    extname: extension,
+    moveToDisk: async (key: string) => {
+      const destination = path.resolve(
+        import.meta.dirname,
+        "..",
+        "..",
+        "storage",
+        key,
+      );
+      fs.writeFileSync(destination, buffer);
+    },
+  };
+
+  const result = await filesService.uploadFile(file as MultipartFile);
+
+  if (result instanceof Error) {
+    throw new Error("File upload failed:", result);
+  }
+
+  return result;
+}
+
+export async function faqScript() {
+  const [articlesResponse, questionsResponse, pivotTableResponse] =
+    await Promise.all([
+      fetch("https://admin.topwr.solvro.pl/items/FAQ_Types"),
+      fetch("https://admin.topwr.solvro.pl/items/FAQ"),
+      fetch("https://admin.topwr.solvro.pl/items/FAQ_Types_FAQ"),
+    ]);
+
+  if (!articlesResponse.ok) {
+    throw new Error(
+      `Failed to fetch articles - got response status code ${articlesResponse.status}`,
+    );
+  }
+  if (!questionsResponse.ok) {
+    throw new Error(
+      `Failed to fetch questions - got response status code ${questionsResponse.status}`,
+    );
+  }
+  if (!pivotTableResponse.ok) {
+    throw new Error(
+      `Failed to fetch pivot table - got response status code ${pivotTableResponse.status}`,
+    );
+  }
+
+  logger.info("Fetching data...");
+  const articlesResult = (await articlesResponse.json()) as GuideArticlesOld;
+  const questionsResult = (await questionsResponse.json()) as GuideQuestionsOld;
+  const pivotTableResult = (await pivotTableResponse.json()) as PivotTable;
+
+  for (const article of articlesResult.data) {
+    let createdAt: DateTime = DateTime.now();
+    let updatedAt = DateTime.fromMillis(0);
+
+    for (const pivot of pivotTableResult.data.filter(
+      (p) => p.FAQ_Types_id === article.id,
+    )) {
+      const question = questionsResult.data.find((q) => q.id === pivot.FAQ_id);
+
+      if (question === undefined) {
+        logger.warn(
+          `Pivot references missing question (ID=${pivot.FAQ_id}) for article ID=${article.id} ("${article.name}"). ` +
+            `This may be a data inconsistency in the source. Skipping this question...`,
+        );
+        continue;
+      }
+
+      if (DateTime.fromISO(question.date_created) < createdAt) {
+        createdAt = DateTime.fromISO(question.date_created);
+      }
+
+      if (DateTime.fromISO(question.date_updated) > updatedAt) {
+        updatedAt = DateTime.fromISO(question.date_updated);
+      }
```

Comment:
minor nitpick: it might be wise to compute the `DateTime` once per object and save it to a temp variable, rather than recomputing it again if the condition is satisfied.